### ISSUE TITLE
CO-OPS offerings are found in the CSW search...

### ISSUE THREAD

**Issue by ocefpaf** [ocefpaf/sscsw#1]
... but no data is downloaded in the pyoos collector step.

**Comment by ocefpaf** [comment]
The latest run has the opposite problem!  The collector does return some data, but the `csw` search does not.  The dates and the bounding box are different in the latest version. (We should pin some dates rather then using `now` for debugging purposes.)

**Comment by emiliom** [comment]
Interesting. I was puzzled about CO-OPS service not being found by the CSW search. That seems to be a pattern with CO-OPS across all/most of your recent "system test" notebooks. I just extended the dates to the previous 30 days (your notebook had 5 days), and it still doesn't show up. Have you managed to make it show up recently? If not, I suggest adding an issue on the [ioos registry](https://github.com/ioos/registry/issues).

**Comment by ocefpaf** [comment]
> Have you managed to make it show up recently?

My SECOORA notebooks still find them, but I will investigate this more..

**Comment by emiliom** [comment]
> My SECOORA notebooks still find them

Really? That's a good sign.

**Comment by ocefpaf** [comment]
Actually no!  I was mistaken!!  See my logs from 2015-12-18:

https://github.com/SECOORA/secoora/blob/gh-pages/notebooks/timeSeries/ssh/2015-12-18/log.txt#L21-L27

Because I forced them later on the notebook I was thinking that everyting was OK :unamused:

**Comment by emiliom** [comment]
FYI, go to the [NGDC Geoportal Browse page](http://www.ngdc.noaa.gov/geoportal/catalog/search/browse/browse.page). On the tree browser on the left, click on NOAA > IOOS. You'll see that all the RA's are there, and have records. Click on NDBC and enter "SOS" as a filter, and it's there. But if you click on NOS (the parent agency for CO-OPS) you'll see that it has 0 records!

**Comment by ocefpaf** [comment]
Indeed!  I guess it is time to raise an issue at ioos registry.

**Comment by ocefpaf** [comment]
I am having a really hard time to finish this notebook.  The notebook cannot find any `NDBC` nor `CO-OPS` data now.

See: http://nbviewer.ipython.org/gist/ocefpaf/07a27754a69bd3c22004

I can hard-code those back again, but that would defeat the purpose of the demo :unamused: 

@emiliom and @rsignell-usgs Can you spot any dump mistake I might be making there?  I tried many different date ranges, including some I was sure got some answers, with no avail.

**Comment by ocefpaf** [comment]
I went back to the SECOORA logs and found out that all 2015 runs but [2015-12-04](https://github.com/SECOORA/secoora/blob/gh-pages/notebooks/timeSeries/ssh/2015-12-04/log.txt#L21-L22)  could not find CO-OPS data via csw search.

**Comment by emiliom** [comment]
What I'm now most puzzled about is that NDBC is not showing up on your search!! I went to the [NGDC GeoPortal](http://www.ngdc.noaa.gov/geoportal/catalog/), and sure enough, there are no records under the IOOS / NDBC collection; [that wasn't the case a week ago!](https://github.com/ocefpaf/sscsw/issues/1#issuecomment-166985320) I'll look into this further in a minute, via a [CSW NDBC collection UUID](https://github.com/ioos/registry/blob/master/uuid.csv) query, and report back. But it looks to me like a new hiccup at NGDC.

The CO-OPS metadata seems to be a more persistent problem. I'll look into it a bit and report back, but let's plan on [creating a new IOOS registry issue.](https://github.com/ioos/registry/issues)

**Comment by ocefpaf** [comment]
> that wasn't the case a week ago! 

The NDBC data exists, see cell [12], but CO-OPS cannot be retrieved with pyoos (cell [13]).

http://nbviewer.ipython.org/gist/ocefpaf/07a27754a69bd3c22004

> The CO-OPS metadata seems to be a more persistent problem. I'll look into it a bit and report back, but let's plan on creating a new IOOS registry issue.

Forgot abou that! Just did it here: https://github.com/ioos/registry/issues/90

**Comment by amilan17** [comment]
I just reset the geoportal harvest for NDBC and NOS/CO-OPs- I think it got out of sync with the recent cleanouts. You should see 1 NDBC record and 11 CO-OPs soon available in the CSW.

**Comment by emiliom** [comment]
That's a huge relief, @amilan17! Thanks for letting us know so quickly. I can see the missing records now on the NGDC Geoportal, including NDBC, CO-OPS (which had been missing for a long time, as @ocefpaf pointed out), and some NANOOS records that were missing today.

**Comment by amilan17** [comment]
Please let me know if there are other missing records - it's probably just the timing of the synchronization in Geoportal.

**Comment by emiliom** [comment]
Yes, most of the problems were likely due to the timing of the synchronization. But the NOS/CO-OPS problem really is/was something different that had been around for weeks, possibly longer. But your cleanup may have fixed it! We'll report back.

**Comment by emiliom** [comment]
CSW queries do indeed return the NDBC and COOPS SOS endpoints now. Things seem to be back to "normal" on the CSW front (or where they were a week ago, with respect to the type of queries we've been running lately).

**Comment by emiliom** [comment]
@ocefpaf: In case this helps, the COOPS SOS record is in fact returned in a CSW query on the OOI bbox and your sst list of names. But it fails with the begin-end dates filter. I'll let you look into that, b/c I don't understand CSW date filtering much.

BTW, I also don't know why the pyoos coops collector may be failing (as you reported earlier today), but that's probably a separate issue. I haven't used the pyoos coops collector, and I don't have time to try it this week. Good luck ...

**Comment by emiliom** [comment]
I'm done with my CSW testing. I've cleaned things up and put all results in [this notebook.](https://github.com/emiliom/stuff/blob/master/IOOS_CSW_NANOOS_and_OOIbbox_tests.ipynb) There, I examined several types of queries relevant to the OOI Endurance bounding box, NANOOS and sst. I also looked more closely at the NOS/CO-OPS SOS CSW record. @ocefpaf, I hope you find something useful there!

**Comment by ocefpaf** [comment]
Thanks @emiliom I will get back to this next year.

(I am trying to wrap up what we have for Kathy's presentation.)
